### PR TITLE
Validate trng fixes #105

### LINKED ISSUE #105: Validate the device RNG.
**Feature description**

Create some `protobuf` messages(maybe compile optionally with a flag) to validate the `rng` in the device.

**Is your feature request related to a problem? Please describe.**
Just to make sure about [the security](https://github.com/skycoin/hardware-wallet/issues/88).

**Describe the solution you'd like**
Be able to run [trezor](https://github.com/trezor/rng-test) and [st](https://www.st.com/content/ccc/resource/technical/document/application_note/4a/6a/82/05/8e/9e/4e/94/DM00073853.pdf/files/DM00073853.pdf/jcr:content/translations/en.DM00073853.pdf) `rng` validations, update the FAQ

**Describe alternatives you've considered**
Make a separate/specific firmware or create a feature for this purpose in this firmware.

**Additional context**
https://github.com/skycoin/hardware-wallet/issues/88

**Possible implementation**
Define a(or some) `protobuf` message, get entropy from the device, save this in a regular pc storage and run statistic tests (like [3.2.2 Second step: NIST statistical test](https://www.st.com/content/ccc/resource/technical/document/application_note/4a/6a/82/05/8e/9e/4e/94/DM00073853.pdf/files/DM00073853.pdf/jcr:content/translations/en.DM00073853.pdf) for example) over this data.

### PR DESCRIPTION
**Pull request by stdevAlDen**
Fixes #105

To be able to run this tests this pr is required: https://github.com/skycoin/hardware-wallet-go/pull/87

Changes:
- Add rules to validate the `TRNG`.
- Make the button confirmation for get entropy optional at build time `DISABLE_BUTTON_CONFIRMATION_TO_GET_ENTROPY`

Does this change need to mentioned in CHANGELOG.md?
yes

Requires changes in protobuff specs?
yes
https://github.com/skycoin/hardware-wallet-protob/issues/32

related to PR skycoin/hardware-wallet-protob# for issue skycoin/hardware-wallet-protob#

Requires testing
no

PD: you can check the attachment as part of the review, but [running the tests in your host is required](https://github.com/simelo/skycoin-hardware-wallet/tree/stdevAlDen_t105_validate_trng#Validate-the-TRNG), please be patient, this can take some hours.

[trng-test.zip](https://github.com/skycoin/hardware-wallet/files/3055048/trng-test.zip)

### CODE REVIEW

**Comment by olemis** [comment]
@stdevalden merge conflicts

File: tiny-firmware/Makefile.include [hunk 1]
```diff
@@ -54,6 +54,8 @@ CFLAGS += -DCONFIDENTIAL='__attribute__((section("confidential")))'
 
 endif
 
+CFLAGS += -DENABLE_BUTTON_CONFIRMATION_TO_GET_ENTROPY=$(ENABLE_BUTTON_CONFIRMATION_TO_GET_ENTROPY)
```
Comment (olemis) [inline tiny-firmware/Makefile.include]: since default build should include button confirmation, name should be `DISABLE_GET_ENTROPY` and inverse logic.

**Comment by olemis** [comment]
@stdevAlDen merge conflicts

**Comment by stdevAlDen** [comment]
> @stdevAlDen merge conflicts

@olemis Done

**Comment by olemis** [comment]
@stdevAlDen Travis failed

File: tiny-firmware/firmware/fsm_impl.c [hunk 2]
```diff
@@ -405,12 +405,33 @@ ErrCode_t msgSetMnemonicImpl(SetMnemonic *msg) {
 	return ErrOk;
 }
 
-ErrCode_t msgGetEntropyImpl(GetEntropy *msg, void (*random_buffer_func)(uint8_t *buf, size_t len)) {
-	RESP_INIT(Entropy);
+ErrCode_t msgGetRawEntropyImpl(
```
Comment (olemis) [inline tiny-firmware/firmware/fsm_impl.c]: wrong approach

Comment (stdevAlDen) [reply tiny-firmware/firmware/fsm_impl.c]: > wrong approach

@olemis how do you recommend me to solve this?

**Pull request merged**